gemma2/gemma2-27b-it-vllm/model/model.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import subprocess
+import uuid
+from vllm import SamplingParams
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+import os
+import huggingface_hub
+
+os.environ["VLLM_ATTENTION_BACKEND"] = "FLASHINFER"
+logger = logging.getLogger(__name__)
+
+
+class Model:
+    def __init__(self, **kwargs):
+        self._config = kwargs["config"]
+        self.model = None
+        self.llm_engine = None
+        self.model_args = None
+        self.hf_secret_token = kwargs["secrets"]["hf_access_token"]
+        os.environ["HF_TOKEN"] = self.hf_secret_token
+        print(
+            "logging in with huggingface authentication token: ", self.hf_secret_token
+        )
+        huggingface_hub.login(token=self.hf_secret_token, add_to_git_credential=True)
```

Comment:
I don't think we need this hub login bit, setting `HF_TOKEN` should be enough (and I think you can then remove `huggingface_hub` altogether from requirements)

Reply:
I tried it without, because we could remove huggingface_hub and git from the requirements, but had issues with vLLM being unable to get the model. Also, by reading hf_secret_token from the truss secret and setting it with os.environ["HF_TOKEN"] we don't need to store the HF token as plaintext in config and can use secrets

Reply:
hmmm, what would logging in to the HF hub accomplish here? I think I was able to deploy this successfully by commenting out L25

Reply:
Might've been an authentication fluke. COnfirmed it still works without login, and also works with autotokenizer for gemma2